synapse/models/infotech.py

```diff
@@ -36,7 +80,160 @@ def cpesplit(text):
     except StopIteration:
         parts.append(part)
 
-    return parts
+    return [part.strip() for part in parts]
+
+# Formatted String Binding characters that need to be escaped
+FSB_ESCAPE_CHARS = [
+    '!', '"', '#', '$', '%', '&', "'", '(', ')',
+    '+', ',', '/', ':', ';', '<', '=', '>', '@',
+    '[', ']', '^', '`', '{', '|', '}', '~',
+    '\\', '?', '*'
+]
+
+FSB_VALID_CHARS = ['-', '.', '_']
+FSB_VALID_CHARS.extend(string.ascii_letters)
+FSB_VALID_CHARS.extend(string.digits)
+FSB_VALID_CHARS.extend(FSB_ESCAPE_CHARS)
+
+def cpe_escape(text):
```

Comment:
Can the cpe functions be renamed or have docstrings added to make it clear if they are expected to be working on data from the various formats ( uris / fsbs / 2.2 / 2.3 ) ? For examples this function doesn't seem friendly for handling a 2.2 URI bound string.